src/middlewared/middlewared/plugins/dyndns.py

```diff
@@ -55,6 +55,23 @@ class Config:
             'ipv6tb@he.net': 'he.net'
         }
 
+    @private
+    async def validate_data(self, data, schema):
+        verrors = ValidationErrors()
+        if data.get('provider') not in await self.provider_choices():
+            verrors.add(
+                f'{schema}.provider',
+                'Please select a valid provider.'
+            )
+
+        if not data.get('domain'):
```

Comment:
Why can't you use empty=False on the List schema?

Reply:
My bad, i overlooked `empty` option for list schema. Thank you for letting me know